src/waterfall-cli.ts

```diff
@@ -218,7 +218,17 @@ export function parse() {
 }
 
 // A helper function provided to commands to keep error messages consistent
-export function error(message: string) {
+export async function error(message: string) {
+	// Allow stdout to flush before proceeding
+	await new Promise<void>((resolve) => {
+		process.stdout.write('', () => {
+			resolve();
+		});
```

Comment:
This should work too, and is a little more succinct:

```suggestion
		process.stdout.write('', resolve);
```

Reply:
I found that the suggested simplification actually gave rise to a compile error.  The Promise used to be defined as returning type of `void`, but with the change, is now going to hand back `Error | undefined'.

The other alternative is to not state what the return type is at all.  TypeScript 3.8.3 does not manage to resolve the response type in that case.  Perhaps TypeScript 3.9.3 will be able to?

This project currently uses TypeScript 3.6.3.